### ISSUE TITLE
[ON HOLD] [Mateusz Ostafil] Should load spinner in cart on remove

### ISSUE THREAD

**Issue by qjimmy** [SAP/spartacus#5009]
## Environment Details

Spartacus:  develop branch

## Steps to Reproduce
- Add one or more items in cart
- Go to your cart
- Remove an item from your cart

## Observed Results

On remove, there is a slight delay that freezes other events. This may cause a diminished user experience if the user tries to click elsewhere during this slight delay.

## Expected Results

Should load spinner while item is being removed from cart.

**Comment by mateuszo** [comment]
Hey,

I need a decision regarding the UX of this feature. There are multiple ways how the spinner could be shown. 

The most standard one is to do it like in add to cart modal or checkout process. However, IMHO it gives a poor user experience here as the whole cart disappears for a moment. Look at a gif below:

![loading](http://g.recordit.co/hPYH5nt9aj.gif)

An alternative approach is to add a spinner as an overlay with backdrop like in this image:

![image](https://user-images.githubusercontent.com/2676966/69424581-4921e600-0d29-11ea-9fd9-3ca22350058d.png)

There are a lot more options so I'm open to suggestions.

**Comment by mszczepaniak** [comment]
I am a fan of this one:
https://blog.angularindepth.com/https-medium-com-thomasburleson-animated-ghosts-bfc045a51fba

**Comment by Xymmer** [comment]
graying is fine for short term, the important thing is to keep the spinner
long term we'd want to do some cool animations
if customer wants to change it, how would they do it?

**Comment by psmul** [comment]
From my point of view, customer could simply overwrite our animations

**Comment by mateuszo** [comment]
Hello again, 

I've investigated this issue in more detail. It turned out that grayed out background with a spinner on top has some drawbacks.

Using `position: absolute;` doesn't look good when we have a longer list of items. See gif below: 
![position-absolute](https://user-images.githubusercontent.com/2676966/69716695-d939a400-110a-11ea-8b8f-480111742be8.gif)

Using `position: sticky;` looks a bit better. However, on mobile, it overflows at the bottom. See the gif:
![position-sticky](https://user-images.githubusercontent.com/2676966/69716877-477e6680-110b-11ea-9d5f-f8098bc2796a.gif)

I've also fiddled with what @mszczepaniak proposed and it looks the best for me:
![ghosts](https://user-images.githubusercontent.com/2676966/69717210-e60ac780-110b-11ea-9f99-e94dd38105aa.gif)
This approach seems to have another benefit. It could be applied to a single row.

**Comment by tobi-or-not-tobi** [comment]
I've actually been refactoring some for the cart related UIs for 2.0, where I also introduced a subtle effect. 

![cart-update-effect](https://user-images.githubusercontent.com/4745807/70141669-dbb27580-1697-11ea-8c64-49e6e7a75534.gif)

Nothing fancy here. The main arguments so far with this approach where:
- make it fairly subtle, I do not like the massive overlays and mega spinners for these small operations
- we need to block all cart operations, but users should still to be able navigate, move away, click the product, etc. 
- we should be very cautious with too much effects. we just increase the cart line quantity, we're not saving the world ;)

What we could improve based on the example I created:
- maybe it's too subtle
- we could do a spinner, but then where? On the price? On the quantity? On the image? 

I had a chat with Anthony on this, who should be the authority to listen to on these things. 

ps. I had to leave the cart related dev for some time, as I was blocked by the #5182 for this. I'll come back to this next week and make a reference to the branch.
